### ISSUE TITLE
Unable to host console app

### ISSUE THREAD

**Issue by VenkateshSrini** [cloudfoundry/dotnet-core-buildpack#163]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?
CLI version 6.30.0 

What version of the buildpack you are using?
1.0.25

If you were attempting to accomplish a task, what was it you were attempting to do?
Hosting a .NET core console application with out getting an error

What did you expect to happen?
The app is expected to start and shutdown grace fully without health check errors

What was the actual behavior?
Health check is failing even after setting to none, process or Port. The CF push command is as below
cf push -u none

Please confirm where necessary:
* [ ] I have included a log output
Yes
* [ ] My log includes an error message
Yes
* [ ] I have included steps for reproduction
1) Push the app it will thrown an error
2) Schedule a Task. The task gets executed and then gets crashed
[pcflogs.txt](https://github.com/cloudfoundry/dotnet-core-buildpack/files/1289661/pcflogs.txt)

**Comment by sclevine** [comment]
Hi @VenkateshSrini,

Can you provide a sample app that demonstrates this behavior?

If you see the health check failing after disabling the health check, it seems likely that the issue is with your foundation and not with the buildpack.

**Comment by sclevine** [comment]
Hi @VenkateshSrini,

There are several issues with the provided task:
1. The .NET Core dependencies require at least 2 GB of disk space. Otherwise the app crashes.
2. The start command has been overridden in manfest.yml and in the Procfile. The buildpack has no way of guaranteeing that the custom start command will work.

If adding more disk doesn't address your issue, can you provide a sample (long-running) app that fails without a custom start command?

**Comment by VenkateshSrini** [comment]
@sclevine ,
Sorry I beg to differ
1) I have increased the .NET core build pack and tested it. But that is not the reason
2) What we are trying to do is to push a console application and not an Http process. But unfortunately the build pack is considering that as Asp.Net core application. To achieve this I have tried to customize start command but that I did not work, Then I tried using the procfile  even that is also what working. 

So it is definitely an .NET build pack issue where it should let us override the start command or provide different build pack for different architecture

**Comment by sclevine** [comment]
The buildpack does not officially support running .NET Core apps as long-running worker processes.

Your app is crashing because the start command you are setting exits immediately. If you want to use the buildpack this way, you'll need to specify a start command that starts a long-running process.

@idoru @dgodd any ideas for a start command that would accomplish this?

**Comment by VenkateshSrini** [comment]
@sclevine ,
Ok. Let me tell you want I'm trying to do. I have basically written this application so that it ca be used as a PCF task. So ideally the application need not run as an LRP and I'm fine with it stopping immediately. 

Given this requirement should we still hold on this process.

**Comment by sclevine** [comment]
If you're trying to run a task via `cf run-task`, what command are you using to run the task?
See: https://docs.pivotal.io/pivotalcf/1-9/devguide/using-tasks.html#run-tasks

It's likely possible to get this working by pushing an app with `--no-start`, staging it without starting it as an LPR, and then using `cf run-task` with a command that runs successfully and exits.

**Comment by sclevine** [comment]
Also worth noting: If you're having trouble with PCF, you can reach out to Pivotal Support at support@pivotal.io.

**Comment by VenkateshSrini** [comment]
@sclevine ,
Thanks for sharing the links. I followed the same link to develop my sample.
If you could see the attached manifest.yml file the no-start is provided.
The pivotal support chat helped me to identify the issue is cause because the asp.net core build pack is not supporting console apps.

**Comment by dgodd** [comment]
Hi @VenkateshSrini I tried your sample and the task ran for me. The key for me was setting the instances to zero, so that no LRPs started.  I have included the steps I used below

```
cd SampleTask
dotnet build
cf push SampleTask --no-start -p bin/Debug/netcoreapp2.0/
cf scale -i 0 SampleTask
cf start SampleTask
cf run-task SampleTask "dotnet SampleTask.dll "Sample"" --name execute-sample-task
## Pause until "cf tasks" notes that all has worked
cf logs SampleTask --recent
```
Note the pause in the above. Tasks Do not immediately start or finish. Using `cf tasks SampleTask` you can wait until the task has finished.

I received the following output as part of `cf logs SampleTask --recent`

```
   2017-10-27T09:40:12.71-0400 [APP/TASK/execute-sample-task/0] OUT Hello Sample!
   2017-10-27T09:40:12.72-0400 [APP/TASK/execute-sample-task/0] OUT Exit status 0
```

**Comment by sclevine** [comment]
Closing due to inactivity, and as @dgodd was able to provide a working solution.

**Comment by VenkateshSrini** [comment]
Hi @dgodd,
Thanks for the steps. When I you do the above steps I would kindly request you to see the logs when the task is completed. It would have reported a crash. Please let me know if you have observed it. If so then I have some research to share with the group.

**Comment by dgodd** [comment]
@VenkateshSrini I'm not sure I understand your question. Are you reporting a bug? asking for a change to this buildpack? or a change to how cloudfoundry tasks work?

**Comment by sclevine** [comment]
@dgodd I think @VenkateshSrini is asking if you see:
```
   2017-09-09T14:54:27.26+0530 [API/0] OUT Process has crashed with type: "web"
```
after
```
   2017-10-27T09:40:12.71-0400 [APP/TASK/execute-sample-task/0] OUT Hello Sample!
   2017-10-27T09:40:12.72-0400 [APP/TASK/execute-sample-task/0] OUT Exit status 0
```
As in the logs provided in the first post. I'm guessing that you don't, because you scaled the web instances down to zero?

**Comment by VenkateshSrini** [comment]
Hi All,
I have found a solution to this problem. The cause is as below
1) earlier there used to be an option to set the health check type as none which has now been removed and is process
2) In .NET core console application the application quits immediately after running. As a result the health check report that as failure even if you give exit(1)
To circumvent that I had used a console.Read() line and then I stop the app as much of my work task based. So this did not left an resource anything dangling. As a result I get graceful shutdown. This is not affected even while running the Task as in Task the expectation is container gets killed once the task is executed.
